ARTstor.js

```diff
@@ -34,98 +37,1043 @@
 	***** END LICENSE BLOCK *****
 */
 
-function getData(doc, checkOnly) {
-	var data = {},
-		widgets = doc.getElementsByClassName('MetaDataWidget'),
-		topWidget = null, topZIndex;
-	
-	// Dtermine the top metadata window
-	for (var i=0; i<widgets.length; i++) {
-		
-		var zindex = doc.defaultView.getComputedStyle(widgets[i], null)
-			.getPropertyValue('z-index');
-		if (!zindex && zindex !== 0) continue;
-		
-		if (!topWidget || topZIndex < zindex) {
-			topWidget = widgets[i];
-			topZIndex = zindex;
-		}
-	}
-	
-	if (!topWidget) return false;
-	
-	var fields = topWidget.getElementsByClassName('field'),
-		found = false;
-	for (var i=0; i<fields.length; i++) {
-		var field = ZU.trimInternal(fields[i].textContent);
-		
-		var value = fields[i].nextElementSibling;
-		if (!value || !value.classList.contains('data')) continue;
-		value = ZU.unescapeHTML(ZU.cleanTags(value.innerHTML));
-		
-		if (!field || !value.trim()) continue;
-		
-		if (checkOnly) return true;
-		
-		found = true;
-		data[field] = value;
-	}
-	
-	return found ? data : false;
-}
-
 function detectWeb(doc, url) {
-	//monitor changes to body's direct children. That's where the metadata popup is added
-	Zotero.monitorDOMChanges(doc.body, {childList: true});
 
-	if (getData(doc, true)) {
-		return "artwork";
-	}
+    if (url.match(/iv2/)) {
+
+        return "artwork";
+    }
+    if (url.match(/imagegroup/) && doc.getElementById("ssContentWrap").style.display == "inline") {
```

Comment:
if `doc.getElementById("ssContentWrap")` returns null, this will throw and everything will break (and we will get an error report), so you need to test to make sure that element exists first.